src/renderer/components/dock/create-resource.tsx

```diff
@@ -21,6 +25,39 @@ interface Props {
 @observer
 export class CreateResource extends React.Component<Props> {
   @observable error = "";
+  templates:GroupSelectOption[] = [];
+
+  componentDidMount(){
+    let rootPaths: any[]= [];
+
+    for (const store of this.templatesStores) {
+      rootPaths = rootPaths.concat(fs.readdirSync(store).map(value => {return {root: store, value};}));
+    }
```

Comment:
```suggestion
    const rootPaths = this.templatesStores
      .flatMap(store => (
        fs.readdirSync(store)
          .map(value => ({ root: store, value })
      ));
```

Reply:
Furthermore, even though this is in `componentDidMount` I would prefer if it was async.